### PR TITLE
Don't render tees or nameplates offscreen

### PR DESCRIPTION
**Pull request by sjrc6**
It seems difficult to measure the fps from the ingame graph or fps counter so these numbers are from NVidia fps overlay

FPS on full multeasymap server, fullscreen, vulkan renderer
/showall 0 - 4500
/showall 1 - 2100
/showall 1 (after preventing offscreen render) - 4200

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [x] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by Jupeyy** [comment]
can we atleast discuss that we use random values?

zooming in will possibly disable nameplates now?

**Comment by def-** [comment]
> zooming in will possibly disable nameplates now?

I don't think so, we use the render positions, so this should show all.

> can we atleast discuss that we use random values?

The border buffer seems to be a guess of how large the tees/nameplates can be. Not sure if there is a good way to get this calculated.

**Review by def- — APPROVED** [review]

**Comment by sjrc6** [comment]
some of the discussion about the buffer happened in discord, for context I based the maximum size on this. 
image

**Comment by Jupeyy** [comment]
zooming in breaks it, just as i expected

**Comment by Jupeyy** [comment]
So just an update, hookline is invisible, tunes make it hard
Will need some more thinking

**Comment by sjrc6** [comment]
hmm I somehow messed up my local libraries

File: src/game/client/components/nameplates.cpp [hunk 1]
```diff
@@ -305,6 +305,16 @@ void CNamePlates::OnRender()
...
 
+	// get screen edges to avoid rendering offscreen
+	float ScreenX0, ScreenY0, ScreenX1, ScreenY1;
+	Graphics()->GetScreen(&ScreenX0, &ScreenY0, &ScreenX1, &ScreenY1);
+	// expand the edges to prevent popping in/out onscreen
+	float BorderBuffer = 400;
+	ScreenX0 -= BorderBuffer;
+	ScreenX1 += BorderBuffer;
+	ScreenY0 -= BorderBuffer;
+	ScreenY1 += BorderBuffer * 2.0f;
```
[the remark this answers is not in the document]
Comment (sjrc6) [reply src/game/client/components/nameplates.cpp:316]: see the picture I posted above

**Comment by sjrc6** [comment]
I wasn't sure what the best way to fix the issue of needing to render hooklines offscreen was. I decided to separate the hookline from the player rendering. If this is too big of a change we could also just render the whole tee when its hookline is active, but most of the performance gain came from not rendering the tees so it would be nice to avoid it.

**Comment by Jupeyy** [comment]
I guess the bounding box should be big enough, I kinda prefer not using random values, but i also currently don't want to work on this.

so ok from my side

**Comment by def-** [comment]
bors r+
Thanks

**Pull request merged**